web/api/v1/api.go

```diff
@@ -781,10 +795,12 @@ func (api *API) remoteRead(w http.ResponseWriter, r *http.Request) {
 		}
 
 		querier, err := api.Queryable.Querier(r.Context(), from, through)
+		remoteReadQueries.Inc()
```

Comment:
This should be done at the beginning of the function. By the way, the metric's update could be delegated to `api.remoteReadGate` as we want to increase the gauge at the beginning of `api.remoteReadGate.Start()` and decrease it when `api.remoteReadGate.End()` returns.

@gouthamve WDYT?

Reply:
this makes sense to me